### ISSUE TITLE
fresh install, looks like pimcore didn't install functions on mysql server

### ISSUE THREAD

**Issue by bjquinn** [pimcore/pimcore#4526]
Fresh install of pimcore on Ubuntu 18.04.  MySQL is on a separate server, and is 8.0.16.  I can log into the admin interface, but certain things result in errors like the below.  When I go to the MySQL server, there are no functions or stored procedures, but the tables and views do exist.

```
Timestamp: Thu Jun 13 2019 17:18:07 GMT-0500 (Central Daylight Time)
Status: 500 | Internal Server Error
URL: /admin/customermanagementframework/segment-assignment/inheritable-segments?id=3&type=document&_dc=1560464273438
Method: GET
Message: An exception occurred while executing 'SELECT PLUGIN_CMF_COLLECT_DOCUMENT_SEGMENT_ASSIGNMENTS(?)' with params [4]:

SQLSTATE[42000]: Syntax error or access violation: 1305 FUNCTION [mydatabase].PLUGIN_CMF_COLLECT_DOCUMENT_SEGMENT_ASSIGNMENTS does not exist
```

**Comment by AlternateIf** [comment]
the error originates from the [CustomerDataFramework](https://github.com/pimcore/customer-data-framework). Have there been any errors installing this framework?

**Comment by bjquinn** [comment]
I didn't see any errors on install.  Is there a log file I should look to for errors with that?

**Comment by bjquinn** [comment]
@AlternateIf thank you for pointing me to the CustomerDataFramework.  @fashxp on talk.pimcore.org also pointed me directly to the 4 SQL files that should run during install at https://github.com/pimcore/customer-data-framework/tree/master/src/Resources/sql/segmentAssignment.

I tried to run these scripts manually -- the datamodel.sql script looks to have run correctly.  The 3 stored function scripts appear not to have run, and would not run on my MySQL server when I tried to run them manually.  I managed to modify them to get them to work.  I had to make two changes: one, do the "delimiter //" hack to get around delimiter issues in the function itself.  And two, I had to add "READS SQL DATA" to the function declaration.  After that, I was able to create the functions correctly on my MySQL server manually, and the errors in the admin interface went away.  An example of one of the modified functions is below.

```
DROP FUNCTION IF EXISTS PLUGIN_CMF_COLLECT_OBJECT_SEGMENT_ASSIGNMENTS;
DELIMITER //
CREATE FUNCTION PLUGIN_CMF_COLLECT_OBJECT_SEGMENT_ASSIGNMENTS(elementIdent INT) RETURNS TEXT
  READS SQL DATA
  BEGIN
    DECLARE segmentIds TEXT;
    DECLARE elementExists TINYINT;
    DECLARE breaks TINYINT;

    SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO segmentIds;
    SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO elementExists;
    SELECT `breaksInheritance` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO breaks;

    WHILE (elementExists = 0 OR breaks IS NULL OR breaks <> 1) AND elementIdent > 1 DO
      SELECT `o_parentId`  FROM `objects` WHERE `o_id` = elementIdent INTO elementIdent;
      SELECT CONCAT_WS(',', segmentIds, (SELECT `segments` FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object')) INTO segmentIds;
      SELECT COUNT(*) FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object' INTO elementExists;
      SELECT `breaksInheritance` INTO breaks FROM `plugin_cmf_segment_assignment` WHERE `elementId` = elementIdent AND `elementType` = 'object';
    END WHILE;

    RETURN segmentIds;
  END//
  DELIMITER ;
```

**Comment by fashxp** [comment]
thx for sharing your results. could you please create a PR here? https://github.com/pimcore/customer-data-framework/pulls

**Comment by bjquinn** [comment]
Thanks.  I created https://github.com/pimcore/customer-data-framework/pull/112 to solve this issue.

**Comment by Jan0707** [comment]
@bjquinn How best to fix this on fresh/exisitng installations that are suffering from missing functions ? Thanks !

**Comment by bjquinn** [comment]
@Jan0707 you can just run the .sql scripts from https://github.com/pimcore/customer-data-framework/tree/master/src/Resources/sql/segmentAssignment manually.  Looks like my pull request has been merged, so you can run those scripts directly.

**Comment by faiyazalam** [comment]
I was installing pimcore ecommerce demo on an Azure web app service. It was not creating any function in the database.
When I debugged this, I found this error message: Error in query (1419): You do not have the SUPER privilege and binary logging is enabled (you *might* want to use the less safe log_bin_trust_function_creators variable).

Then I changed the parameter log_bin_trust_function_creators = ON and then I reinstalled the pimcore ecommerce demo. It worked.
